internal/ioutil/io_test.go

```diff
@@ -50,3 +54,62 @@ func TestUnwrapNopCloser(t *testing.T) {
 		})
 	}
 }
+
+func TestCopyBuffer(t *testing.T) {
+	type args struct {
+		src  io.Reader
+		buf  []byte
+		desc ocispec.Descriptor
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantDst string
+		wantErr error
+	}{
+		{
+			name:    "exact buffer size, no errors",
+			args:    args{bytes.NewReader([]byte("foo")), make([]byte, 3), content.NewDescriptorFromBytes("test", []byte("foo"))},
+			wantDst: "foo",
+			wantErr: nil,
+		},
+		{
+			name:    "small buffer size, no errors",
+			args:    args{bytes.NewReader([]byte("foo")), make([]byte, 1), content.NewDescriptorFromBytes("test", []byte("foo"))},
+			wantDst: "foo",
+			wantErr: nil,
+		},
+		{
+			name:    "big buffer size, no errors",
+			args:    args{bytes.NewReader([]byte("foo")), make([]byte, 5), content.NewDescriptorFromBytes("test", []byte("foo"))},
+			wantDst: "foo",
+			wantErr: nil,
+		},
+		{
+			name:    "wrong digest",
+			args:    args{bytes.NewReader([]byte("foo")), make([]byte, 3), content.NewDescriptorFromBytes("test", []byte("bar"))},
+			wantDst: "foo",
+			wantErr: content.ErrMismatchedDigest,
+		},
+		{
+			name:    "wrong size",
+			args:    args{bytes.NewReader([]byte("foo")), make([]byte, 3), content.NewDescriptorFromBytes("test", []byte("fo"))},
+			wantDst: "foo",
+			wantErr: content.ErrTrailingData,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dst := &bytes.Buffer{}
+			err := CopyBuffer(dst, tt.args.src, tt.args.buf, tt.args.desc)
+			if err != tt.wantErr {
+				t.Errorf("CopyBuffer() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			gotDst := dst.String()
+			if err == nil && gotDst != tt.wantDst {
```

Comment:
Do we need to check `err == nil`?

Reply:
After discussion, we think we need this check.